compiler/noirc_frontend/src/tests/name_shadowing.rs

```diff
@@ -0,0 +1,421 @@
+#[cfg(test)]
+mod name_shadowing_tests {
```

Comment:
The extra mod here seems unnecessary when this is already in the tests super module.

You can use `#![cfg(test)]` instead